packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -676,14 +682,18 @@ function normalizeToEnvName(input: string): string {
   return input.toUpperCase().replace(/-/g, "_").replace(/\./g, "__"); // replace "-" to "_" and "." to "__"
 }
 
-function generateResourceName(ctx: SolutionContext): void {
+function generateResourceBaseName(appName: string, envName: string): string {
   const maxAppNameLength = 10;
-  const appName = ctx.projectSettings!.appName;
-  const suffix = ctx.envInfo.profile.get(GLOBAL_CONFIG)?.getString("resourceNameSuffix");
   const normalizedAppName = appName.replace(/[^a-zA-Z0-9]/g, "").toLowerCase();
+  return normalizedAppName.substr(0, maxAppNameLength) + envName.substr(0, maxAppNameLength);
```

Comment:
for `envName.substr(0, maxAppNameLength);`, use another variable for the max length? so we can tweak the 2 values separately.

Reply:
sure